libraries/tock-register-interface/src/fields.rs

```diff
@@ -413,6 +413,12 @@ macro_rules! register_bitmasks {
                     }
                 }
             }
+
+            impl From<Value> for FieldValue<$valtype, $reg_desc> {
+                fn from(v: Value) -> Self {
+                    Self::new($crate::bitmask!($numbits), $offset, v as $valtype)
```

Comment:
This looks good to me as is, but can we perhaps instead just return the in-scope named constant here instead? That might make it more obvious that this is converting to the constant FieldValue and reduces code duplication, as these macro interactions are already sufficiently complicated. :)

Reply:
Do you mean adding a match expression with an arm for each enum value, returning the corresponding named constant? The seems like more code duplication, and makes for a more complicated macro than just doing the cast here.

Reply:
Oh yeah, sorry, I've mistaken the types here. This seems to be as good a solution as we can get.